### PR TITLE
feat: Integrate /groups/%id with RBAC checks

### PR DESCRIPTION
**Pull request by gkarat**
Implements https://issues.redhat.com/browse/ESSNTL-4937, https://issues.redhat.com/browse/ESSNTL-4938, https://issues.redhat.com/browse/ESSNTL-4940, https://issues.redhat.com/browse/ESSNTL-4941.

Users need to have at least the read permissions in order to see the group name, its systems and access the group info tab. Also, the write permissions are required in order to delete or rename a group.

## How to test

You can try to use the account that has only inventory:groups:read or inventory:groups:write.

### CODE REVIEW

File: src/components/InventoryGroupDetail/GroupDetailHeader.js [hunk 1]
```diff
@@ -21,20 +21,33 @@ import PropTypes from 'prop-types';
...
+  );
+
   const [dropdownOpen, setDropdownOpen] = useState(false);
   const [renameModalOpen, setRenameModalOpen] = useState(false);
   const [deleteModalOpen, setDeleteModalOpen] = useState(false);
 
   const name = data?.results?.[0]?.name;
   const title =
-    uninitialized || loading ? (
+    canRead && (uninitialized || loading) ? (
```
Comment (mkholjuraev) [inline src/components/InventoryGroupDetail/GroupDetailHeader.js:50]: This seems a bit off. If canRead is falsy, I suspect the title will be equal to name or groupId?  I believe, one if-else would make this comparison easier to understand the logic even if it is really basic and not trendy.

File: src/components/InventoryGroupDetail/InventoryGroupDetail.js [hunk 2]
```diff
@@ -21,9 +25,30 @@ const InventoryGroupDetail = ({ groupId }) => {
   const chrome = useChrome();
   const groupName = data?.results?.[0]?.name;
 
+  const { hasAccess: canView } = usePermissionsWithContext(
+    REQUIRED_PERMISSIONS_TO_READ_GROUP(groupId)
+  );
+
+  const EmptyStateNoAccess = () => (
```
Comment (mkholjuraev) [inline src/components/InventoryGroupDetail/InventoryGroupDetail.js:32]: Let's refactor this empty state outside of this component. Creating a dedicated folder for such prone-to-be-shared components would be great.

**Review by mkholjuraev — COMMENTED** [review]
Codewise looks good, a few comments though. Also, as I am not actively following groups UX guidelines, I focused on the code itself.

File: src/components/InventoryGroupDetail/GroupDetailHeader.js (hunk 1 above)
Comment (gkarat) [reply src/components/InventoryGroupDetail/GroupDetailHeader.js]: good catch, there was a bug, yes. revisited in the last commit

File: src/components/InventoryGroupDetail/InventoryGroupDetail.js (hunk 2 above)
Comment (gkarat) [reply src/components/InventoryGroupDetail/InventoryGroupDetail.js]: done!:)

**Review by mkholjuraev — APPROVED** [review]
Codewise LGTM!

**Pull request merged**